helix-view/src/editor.rs

```diff
@@ -94,16 +94,17 @@ impl Editor {
         self._refresh();
     }
 
-    pub fn set_theme_from_name(&mut self, theme: &str) {
+    pub fn set_theme_from_name(&mut self, theme: &str) -> anyhow::Result<()> {
         let theme = match self.theme_loader.load(theme.as_ref()) {
             Ok(theme) => theme,
             Err(e) => {
                 log::warn!("failed setting theme `{}` - {}", theme, e);
-                return;
+                anyhow::bail!("failed setting theme `{}` - {}", theme, e);
             }
         };
 
         self.set_theme(theme);
+        Ok(())
```

Comment:
If you log at the call site as suggested above, I see this becoming as simple as
```rust
let theme = self.theme_loader.load(theme.as_ref()).context(|| format!("failed setting theme `{}`", theme))?;
self.set_theme(theme);
Ok(())
```